### PR TITLE
multi controlnet

### PR DESCRIPTION
**Pull request by williamberman**

### CODE REVIEW

File: controlnet.md [hunk 1]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/controlnet/anime_do_yoga.png" width=600/>
 </p>
 
+### Combining multiple conditionings
+
+Multiple controlnet conditionings can be conbined for a single image generation. Pass a list of controlnets to the pipeline's constructor and a corresponding list of conditionings to `__call__`.
```
Comment (sayakpaul) [inline controlnet.md:403]:
```suggestion
Multiple ControlNet conditionings can be combined for a single image generation. Pass a list of ControlNets to the pipeline's constructor and a corresponding list of conditionings to `__call__`.
```

File: controlnet.md [hunk 2]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+
+When combinding conditionings, it is helpful to mask conditionings such that they do not overlap. In the example, we mask the middle of the canny map where the pose
```
Comment (sayakpaul) [inline controlnet.md:405]: I think we should explain it a bit more with some visuals.

File: controlnet.md [hunk 3]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+<p align="center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/landscape.png" width=600/>
+</p>
+
+
+#### Person
+
+<p align="center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/person.png" width=600/>
+</p>
```
Comment (sayakpaul) [inline controlnet.md:421]: What are these? Would be helpful to the users to know what these are.

File: controlnet.md [hunk 4]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+
+openpose = OpenposeDetector.from_pretrained("lllyasviel/ControlNet")
+
+openpose_image = load_image(
+    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/person.png"
+)
+openpose_image = openpose(openpose_image)
+
+
+
```
Comment (sayakpaul) [inline controlnet.md:471]:
```suggestion
# extract poses with the OpenPose model
openpose = OpenposeDetector.from_pretrained("lllyasviel/ControlNet")
openpose_image = load_image(
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/person.png"
)
openpose_image = openpose(openpose_image)

```

File: controlnet.md [hunk 5]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+# create pipeline
+
```
Comment (sayakpaul) [inline controlnet.md:473]:
```suggestion
```

File: controlnet.md [hunk 6]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+#### Masked Landscape Conditioning
+
+<p align="center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/controlnet/landscape_canny_masked.png" width=600/>
+</p>
+
+
 Throughout the examples, we explored multiple facets of the [`StableDiffusionControlNetPipeline`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/stable_diffusion/controlnet) to show how easy and intuitive it is play around with ControlNet via Diffusers. However, we didn't cover all types of conditionings supported by ControlNet. To know more about those, we encourage you to check out the respective model documentation pages:
 
 * [lllyasviel/sd-controlnet-depth](https://huggingface.co/lllyasviel/sd-controlnet-depth)
```
Comment (sayakpaul) [inline controlnet.md:524]: Wrong link, I guess? It just shows the Depth ControlNet model. Maybe you missed out on it?

File: controlnet.md [hunk 7]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+
+<p align="center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/controlnet/person_pose.png" width=600/>
+</p>
+
+#### Masked Landscape Conditioning
+
+<p align="center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/blog/controlnet/landscape_canny_masked.png" width=600/>
+</p>
```
Comment (sayakpaul) [inline controlnet.md:519]: These two things read a bit incoherent w.r.t the content presented just before.

**Review by sayakpaul — COMMENTED** [review]
Thanks! Left some nits.

File: controlnet.md (hunk 2 above)
Comment (patrickvonplaten) [reply controlnet.md:405]: Ah I didn't know that this is possible! Cool, agree that visuals would be nice, but not a must for me to merge this.

File: controlnet.md [hunk 8]
```diff
@@ -398,6 +398,127 @@ image_grid(output.images, 2, 2)
...
+    prompt,
+    images,
+    num_inference_steps=20,
+    generator=generator,
+    negative_prompt=negative_prompt,
+    controlnet_conditioning_scale=[1.0, 0.8],
+).images[0]
+
+image.save("./multi_controlnet_output.png")
+```
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply controlnet.md:507]: Ah it's been shown above, could we maybe change the order a bit:

First show two original images.
Then show code how to process them
Then show processed images
Then show controlnet generation code
Then show resulting image

**Review by yiyixuxu — APPROVED** [review]
super cool! 
agree that we should show the images same order as they are processed and also show them immediately after they are processed 

do we have to mask images manually like this? 😂😂😂 any tool we can use?

File: controlnet.md (hunk 6 above)
Comment (williamberman) [reply controlnet.md:523]: I don't follow?

File: controlnet.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (williamberman) [reply controlnet.md:513]: done!

File: controlnet.md (hunk 3 above)
Comment (williamberman) [reply controlnet.md]: I don't follow?

**Comment by williamberman** [comment]
> super cool! agree that we should show the images same order as they are processed and also show them immediately after they are processed
> 
> do we have to mask images manually like this? 😂😂😂 any tool we can use? (would be nice to add a link or something if we have one!)

There are tools online that you can use, but I found manually zeroing out the image to be easier for the basic example. One line of code vs uploading and downloading something

File: controlnet.md (hunk 2 above)
Comment (williamberman) [reply controlnet.md]: Assuming we're talking about the masking, I think the canny conditioning with the middle zeroed out is sufficient

**Review by patrickvonplaten — COMMENTED** [review]
Nice let's maybe now quickly add credit for the author and we should be good

File: controlnet.md (hunk 6 above)
Comment (sayakpaul) [reply controlnet.md:522]: Oh sorry. Please disregard.

**Pull request merged**